File: backend/app/schemas/avaliacao.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class AvaliacaoPedidoBase(BaseModel):
    id_pedido: str
    avaliacao: int
    titulo_comentario: Optional[str] = None
    comentario: Optional[str] = None
    data_comentario: Optional[datetime] = None
    data_resposta: Optional[datetime] = None


class AvaliacaoPedidoCreate(AvaliacaoPedidoBase):
    pass


class AvaliacaoPedidoResponse(AvaliacaoPedidoBase):
    id_avaliacao: str

    model_config = ConfigDict(from_attributes=True)


class AvaliacaoResponse(AvaliacaoPedidoResponse):
    pass

File: frontend/src/types/avaliacao.ts
export type AvaliacaoResponse = {
  id_avaliacao: string
  id_pedido: string
  avaliacao: number
  titulo_comentario: string | null
  comentario: string | null
  data_comentario: string | null
  data_resposta: string | null
}

export type AvaliacoesProdutoResponse = {
  media: number
  total_avaliacoes: number
  avaliacoes: AvaliacaoResponse[]
}
